docs/Way-of-Working/UbiquitousLanguage.md

```diff
@@ -0,0 +1,227 @@
+their similarity by some common characteristics. Unlike a classification problem, these groups (called clusters) are not
+predefined by humans, but identified by machine learning algorithms while learning the input data. The elements in each
+cluster are similar among themselves and different from all the others. Common clustering algorithms are k-means,
+hierarchical clustering, spectral clustering, etc..
+
+### Confusion Matrix
+
+A confusion matrix is a table illustrating the predictive performance of a classification model. Usually, a confusion
+matrix is created for a binary output (i.e., prediction problems with only two types of prediction—e.g., whether a
+transaction is fraudulent or not), so the resulting table is a two by two table. A confusion matrix represents the
+relationships between predictions vs. actual labels for both classes. It easily surfaces the amount of accurate
+predictions made (true positives and true negatives), and the amount of false positives (type I error) and false
+negatives (type II error) made.
+
+### Correlation
+
+Correlation is the strength and direction of the relationship between two or more variables, measured by a correlation
+coefficient, or Pearson coefficient. Statistically, a correlation coefficient is the ratio of
+the [covariance](https://www.datacamp.com/blog/data-science-glossary#covariance) of two variables to the product of
+their standard deviations. It can take the values from -1 (a perfect negative correlation) to 1 (a perfect positive
+correlation). The presence of correlation between two variables does not imply cause-effect relation.
+
+### Covariance
+
+Covariance is a measure of the relationship between two variables.
+Unlike [variance](https://www.datacamp.com/blog/data-science-glossary#variance) that measures variations inside the same
+variable, covariance shows how variations in one variable influence changes in the second one. Covariance is used for
+calculating a correlation coefficient.
+
+### Cross-Validation
+
+Cross-validation is a resampling method when training machine learning models that splits labeled data into training and
+test sets. In each iteration of cross validation, different parts of the data are used to train and test the model. The
+training set is used to train a model, and the test set is used to make predictions and compare them with the actual
+labels for those entries. Afterwards, an overall accuracy metric is calculated to estimate the predictive performance of
+the resulting model.
+
+### Data Enrichment
+
+Data enrichment is the processes of enhancing, refining, and augmenting raw data, to make it more useful for the
+organization and, as a result, to obtain more meaningful business insights and optimize predictive analytics.
+
+### Dataframe
+
+A dataframe is the tabular data structure with labeled axes (rows and columns) of potentially different types.
```

Comment:
I think the term dataframe is specific to the Pandas package. Maybe good to mention that here.